docs/content/concepts/assets/multi-assets.mdx

```diff
@@ -0,0 +1,114 @@
+```
+
+By default, the names of the outputs will be used to form the asset keys of this multi-asset. The decorated function will be used to create the op for these assets, and must emit an output for each of them. In this case, we can emit multiple outputs by returning a tuple of values, one for each asset.
+
+### Customizing how assets are materialized with IO managers
+
+As with regular assets, you can customize how each asset is materialized with [IO managers](/concepts/io-management/io-managers). You may specify an `io_manager_key` on each output of the multi-asset.
+
+```python file=/concepts/assets/multi_assets.py startafter=start_io_manager_multi_asset endbefore=end_io_manager_multi_asset
+from dagster import multi_asset, Out
+
+
+@multi_asset(
+    outs={
+        "s3_asset": Out(io_manager_key="s3_io_manager"),
+        "adls_asset": Out(io_manager_key="adls2_io_manager"),
+    },
+)
+def my_assets():
+    return "store_me_on_s3", "store_me_on_adls2"
+```
+
+### Subsetting multi-assets
+
+By default, it is assumed that the computation inside of a multi-asset will always produce the contents all of the associated assets. This means that attempting to execute a set of assets that produces some, but not all of the assets defined by a given multi-asset will result in an error.
+
+Sometimes, the underlying computation is sufficiently flexible to allow for computing an arbitrary subset of the assets that it is associate with. For these cases, you may set the `is_required` attribute of the outputs to `False`, and set the `can_subset` parameter of the decorator to `True`.
+
+Inside the body of the function, we can use `context.selected_output_names` or `context.selected_asset_keys` to find out which computations should be run.
+
+```python file=/concepts/assets/multi_assets.py startafter=start_subsettable_multi_asset endbefore=end_subsettable_multi_asset
+from dagster import multi_asset, Out, Output
+
+
+@multi_asset(
+    outs={
+        "a": Out(is_required=False),
+        "b": Out(is_required=False),
+    },
+    can_subset=True,
+)
+def split_actions(context):
+    if "a" in context.selected_output_names:
+        yield Output(value=123, output_name="a")
+    if "b" in context.selected_output_names:
+        yield Output(value=456, output_name="b")
+```
+
+Because our outputs are now optional, we can no longer rely on returning a tuple of values, as we don't know in advance which values will be computed. Instead, we explicitly `yield` each output that we're expected to create.
+
+### Dependencies inside multi-assets
+
+When a multi-asset is created, it is assumed that each output asset depends on all of the input assets. However, this may not always be the case. In these situations, you may optionally provide a mapping from each output asset to the set of <PyObject object="AssetKey" />s that it depends on. This information will be used when displaying lineage information in Dagit and when parsing selections over your asset graph.
```

Comment:
```suggestion
When a multi-asset is created, it is assumed that each output asset depends on all of the input assets. This may not always be the case, however.

In these situations, you may optionally provide a mapping from each output asset to the set of <PyObject object="AssetKey" />s that it depends on. This information is used to display lineage information in Dagit and for parsing selections over your asset graph.
```

This paragraph is on the longer side and is also informationally dense - let's break it up a bit. I also changed some of this to use active voice. Feel free to re-write!